### PR TITLE
Add touch support to Image Editor Crop

### PR DESCRIPTION
**Pull request by lamosty**
Adds touch support to `Draggable` component. Part of #8044.

## Testing

1. Either use a mobile device (with touch support) or simulate it with dev tools;
2. Navigate to Image Editor and try to use crop feature;
3. Try to move the crop box;
4. Is it all working fine? No strange things?

## Todo:

- [x] Squash commits and write nicer messages (not "WIP") @lamosty
- [x] Test on iPhone @artpi
- [x] Test on Android @lamosty
- [ ] ~~Test on MS devices (will probably not work as they use `Pointer` events)~~
- [x] Show Edit Image button for touch/mobile devices @lamosty
- [x] Bring back Edit image button @artpi 

/cc @gwwar @artpi @obenland @retrofox @adambbecker

### CODE REVIEW

File: client/post-editor/media-modal/detail/detail-item.jsx [hunk 1]
```diff
@@ -48,6 +48,7 @@ module.exports = React.createClass( {
 	},
 
 	isMobileTouchDevice() {
+		return false;
```
Comment (gwwar) [inline client/post-editor/media-modal/detail/detail-item.jsx]: Probably better to remove `isMobileTouchDevice` entirely, right?

Comment (lamosty) [inline client/post-editor/media-modal/detail/detail-item.jsx]: Yep, I got it in my TODO list in the PR's description. Sorry for the mess.

**Comment by gwwar** [comment]
👍 Tested this on a native android device. Works really well. 

If you rebase we recently moved the "Edit Image" button to on top of the image on the details page. In mobile, we'll want the button back in the top right.

**Comment by gwwar** [comment]
Another thing is that gestures would be nice to implement, but we can wait for another iteration.

**Comment by artpi** [comment]
Tested on iPhone.
You may want to rebase

img_3143 3

Since Mobile Safari is still Safari, that is probably the same issue as https://github.com/Automattic/wp-calypso/pull/8158

That got merged 2 hrs ago, so you may want to rebase to pull that fix

**Comment by artpi** [comment]
Also : in https://github.com/Automattic/wp-calypso/pull/8210 I may have moved the button to the image preview
But we dont show image preview on mobile.
We want the old place on mobile, I added a task for myself at the top post

**Comment by artpi** [comment]
I rebased your PR to pick up fixes for safari and I added some responsive black magic to have old edit button behaviour on mobile.

zrzut ekranu 2016-09-27 o 19 43 24
zrzut ekranu 2016-09-27 o 19 42 53

I also tested on iPhone. 

After I zoom in, the zoom-out part is kind of annoying and I recall also @obenland complaining about that

I have a nice movie which I cant upload because my internet is shit.

**Comment by lamosty** [comment]
@artpi: (probably) after your changes, I don't see the Edit Image button in portrait mode on my LG G2 phone. Only when I switch to landscape it appears. Seems like it's off-screen. Can you look into this, please?

**Comment by lamosty** [comment]
I was not able to test this on MS touch devices because I don't have any and it seems like IE11 doesn't have touch device simulation like Chrome does (tried this in my Win 8.1 VM).

Knowing this, can we merge this anyways? We should be still better off with this changes in than not having them. Worst case scenario is that it won't be usable on MS devices which it already isn't at the moment. What is more, not many of our users use MS devices.

What do you think @gwwar?

**Comment by gwwar** [comment]
We're pretty close here. The MS touch stuff we can leave for now.

Before 🚢, let's do the following:
1) Minor, but it'd be nice to match the breakpoint where we go to one column. There are a few widths that I can trigger this:

screen shot 2016-09-29 at 3 31 59 pm

2) We have a js error that fires in the steps described below. I'm not sure if this is fixed in other branches.

- Open item details view
- Click Edit Image
- Resize cropper and move it around
- Press cancel
- Resize the window

**Comment by lamosty** [comment]
> Minor, but it'd be nice to match the breakpoint where we go to one column. There are a few widths that I can trigger this:

I can reproduce the bad behavior with Chrome DevTools Device Toolbar toggled on (desktop mode with 100% zoom) as shown on the screenshot below.

selection_030

However, I'm not able to reproduce that if I close the Device Toolbar at the same screen width:

selection_031

I've also recorded a short video in which I change screen width and everything works fine: https://drive.google.com/file/d/0B8e-IpRwSdjtYnJrdGZrLTA5T2M/view.

Maybe it's just Chrome DevTools Device Toolbar bug?

**Comment by lamosty** [comment]
> We have a js error that fires in the steps described below. I'm not sure if this is fixed in other branches.

I'm not able to reproduce. Video: https://drive.google.com/file/d/0B8e-IpRwSdjtMEduTHVKa1FHNEU/view.

I've rebased recently so that might have helped. Could you try to get the error again, please?

**Comment by lamosty** [comment]
I retested this on my LG G2 phone after getting #8296 in and now it's pretty usable. The crop box movement and resizing is very responsive, even on larger images.

**Comment by gwwar** [comment]
Nice, looks like the js error was fixed from the rebase.

Re the button, there's a vertical aspect to this as well, plus the aspect ratio of the img. This is pretty minor though so we can maybe clean this up later in another PR:
resize

**Comment by lamosty** [comment]
Thanks for the reviews @gwwar!

Re the button: we would have to use custom media queries because the media modal is in two columns only in `>893px` screen width. But we don't like custom media queries (see [CSS guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md)).

@mtias do you have an idea about the above? :)

**Pull request merged**